library/src/scala/tasty/reflect/Core.scala

```diff
@@ -306,6 +306,9 @@ trait Core {
       /** Type tree representing a type binding */
       type Bind <: TypeTree
 
+      /** Type tree within a block with aliases `{ type U1 = ... ; T[U1, U2] }` */
+      type Block <: TypeTree
+
```

Comment:
Currently only in reified quoted code to splice in the types trees into typedefs rather than reconstructing the full type tree to then splice the type trees. The are some potential other use case.